osu.Game/Rulesets/Objects/HitObject.cs

```diff
@@ -89,14 +87,21 @@ protected virtual void ApplyDefaultsToSelf(ControlPointInfo controlPointInfo, Be
             Kiai = effectPoint.KiaiMode;
             SampleControlPoint = samplePoint;
 
-            overallDifficulty = difficulty.OverallDifficulty;
-            hitWindows = null;
+            if (HitWindows == null)
+                HitWindows = CreateHitWindows();
+            HitWindows?.SetDifficulty(difficulty.OverallDifficulty);
```

Comment:
should `CreateHitWindows` actually return null?

Reply:
Good point. HitWindows can/should be nullable though (catch).